drivers/sensor/max17262/max17262.c

```diff
@@ -0,0 +1,342 @@
+		break;
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		/* Get time in ms */
+		if (data->time_to_empty == 0xffff) {
+			valp->val1 = 0;
+			valp->val2 = 0;
+		} else {
+			tmp = data->time_to_empty * 5625;
+			convert_millis(valp, tmp);
+		}
+		break;
+	case SENSOR_CHAN_GAUGE_TIME_TO_FULL:
+		if (data->time_to_full == 0xffff) {
+			valp->val1 = 0;
+			valp->val2 = 0;
+		} else {
+			/* Get time in ms */
+			tmp = data->time_to_full * 5625;
+			convert_millis(valp, tmp);
+		}
+		break;
+	case SENSOR_CHAN_GAUGE_CYCLE_COUNT:
+		valp->val1 = data->cycle_count / 100;
+		valp->val2 = data->cycle_count % 100 * 10000;
+		break;
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		convert_millis(valp, data->design_cap);
+		break;
+	case SENSOR_CHAN_GAUGE_DESIGN_VOLTAGE:
+		convert_millis(valp, config->design_voltage);
+		break;
+	case SENSOR_CHAN_GAUGE_DESIRED_VOLTAGE:
+		convert_millis(valp, config->desired_voltage);
+		break;
+	case SENSOR_CHAN_GAUGE_DESIRED_CHARGING_CURRENT:
+		valp->val1 = data->ichg_term;
+		valp->val2 = 0;
+		break;
+	case MAX17262_COULOMB_COUNTER:
+		/* Get spent capacity in mAh */
+		data->coulomb_counter = 0xFFFF - data->coulomb_counter;
+		valp->val1 = data->coulomb_counter / 2;
+		valp->val2 = data->coulomb_counter % 2 * 10 / 2;
+		break;
+	default:
+		LOG_ERR("Unsupported channel!");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Read register values for supported channels
+ *
+ * @param dev MAX17262 device to access
+ * @return 0 if successful, or negative error code from I2C API
+ */
+static int max17262_sample_fetch(const struct device *dev,
+				 enum sensor_channel chan)
+{
+	struct max17262_data *data = dev->data;
+	struct {
+		int reg_addr;
+		int16_t *dest;
+	} regs[] = {
+		{ VCELL, &data->voltage },
+		{ AVG_CURRENT, &data->avg_current },
+		{ ICHG_TERM, &data->ichg_term },
+		{ REP_SOC, &data->state_of_charge },
+		{ INT_TEMP, &data->internal_temp },
+		{ REP_CAP, &data->remaining_cap },
+		{ FULL_CAP_REP, &data->full_cap },
+		{ TTE, &data->time_to_empty },
+		{ TTF, &data->time_to_full },
+		{ CYCLES, &data->cycle_count },
+		{ DESIGN_CAP, &data->design_cap },
+		{ COULOMB_COUNTER, &data->coulomb_counter },
+	};
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+	for (size_t i = 0; i < ARRAY_SIZE(regs); i++) {
+		int rc;
+
+		rc = max17262_reg_read(dev, regs[i].reg_addr, regs[i].dest);
+		if (rc != 0) {
+			LOG_ERR("Failed to read channel %d", chan);
+			return rc;
+		}
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Initialise the fuel gauge
+ *
+ * @param dev MAX17262 device to access
+ * @return 0 for success
+ * @return -EINVAL if the I2C controller could not be found
+ */
+static int max17262_gauge_init(const struct device *dev)
+{
+	struct max17262_data *data = dev->data;
+	const struct max17262_config *const config = dev->config;
+	int16_t tmp, hibcfg;
+
+	data->i2c = device_get_binding(config->i2c_bus);
+	if (!data->i2c) {
+		LOG_ERR("Could not get pointer to %s device", config->i2c_bus);
```

Comment:
With the device pointer in the config structure, this should be: 

```suggestion
	if (!device_is_ready(config->i2c)) {
		LOG_ERR("Could not get pointer to %s device", config->i2c->name);
```

And `data` becomes an unused variable which can be removed.

Reply:
Implemented.